model/src/main/java/com/javahelp/model/user/UserPassword.java

```diff
@@ -1,63 +1,62 @@
-package com.javahelp.model.userEntityTest;
+package com.javahelp.model.user;
 
 import java.util.Base64;
 
+/**
+ * A class for the password of a user entity
+ */
 public class UserPassword {
+
     /**
-     * A class for hashing a user's password
      * --- Attributes ---
      * salt - a current salt of a user
      * hash - the current and hashed password of the user
-     * user - the user object
      */
-
     private byte[] salt;
 
     private byte[] hash;
 
-    private User user;
-
     /** The constructor to build an instance of the password of a user account.
      *
      * @param salt - The currently used salt for the hashed password.
      * @param hash - The already-hashed password
-     * @param user - the current user account
      */
-
-    public UserPassword(User user, byte[] salt, byte[] hash){
-        this.user = user;
+    public UserPassword(byte[] salt, byte[] hash){
         this.salt = salt;
         this.hash = hash;
     }
```

Comment:
Now that I think about it, we may also want a constructor that takes in a `String` and a salt, and hashes the string to create a new `UserPassword`. This may be outside of the scope of this PR for now, so perhaps it should be left alone for the time being.

Reply:
Actually now that I think about it, a constructor such as the one that currently exists will most likely never be used. We will need a constructor that hashes a `String` eventually. Maybe for now, it's best to leave this without any constructor.